synapse/server.py

```diff
@@ -308,7 +308,10 @@ def build_initial_sync_handler(self):
         return InitialSyncHandler(self)
 
     def build_profile_handler(self):
-        return ProfileHandler(self)
+        if self.config.worker_app:
```

Comment:
could we have this return `WorkerProfileHandler` and then override it in `synapse.server.HomeServer`? possibly with a comment here to say that's what's going on?

Reply:
hrm, ok, let's punt it for now, though I'm not sure I agree.

Reply:
Certainly something we can revisit if/when we try and clean up the worker split out